freva-rest/src/databrowser_api/endpoints.py

```diff
@@ -235,3 +243,78 @@ async def load_data(
         status_code=status_code,
         media_type="text/plain",
     )
+
+
+@app.put(
```

Comment:
Nitpck: this should be post.

Reply:
I'm good with changing this to `post` but can i ask what ur reason is? Since both methods are going to do a same job, in principal switching to `post` doesn't change anything in the functionality and final result. So no worries about technical part. But i read a bit about these CRUD methods and restful api principals, and it seems `put` is idempotent, whihc means, multiple same req has a same effect, or it might update if there is any changes. And in this case, add-user-data, user can add/update same data as many as wishes, without any effect. But from another point of view, i think u r right! because we call it specifically `add`-user-data, so it's makes more sense to call it `post`. i've already changed to `post`, but plz let me know what ur concern is regarding changing the method ....

Reply:
POST is used when the operation involves submitting data to be processed, which is the case when users are adding or indexing new data. This means the server is doing some processing (like adding the data to the index), and the data being sent may result in a new resource or trigger a specific action. 

If, on the other hand, the index location would be known then PUT would be appropriate. But since we cannot tell this I'd go for POST. Also, apache solr always uses POST, even for delete actions. 



If I think about it, you probably can make the delete methods also POST.